### DISCUSSION TITLE
regular expressions

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#784] [Help]
Where can I find information about regular expressions and how to use them in Freeplane?

Thanks.

~ Bal

**Comment by macmarrum** [comment]
The topic of regular expressions is vast. It's probably best to use one of the tutorials available online. https://www.regular-expressions.info/ looks good to me.

I often use the "Summary of regular-expression constructs" in Java as a quick reference: https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#sum

Freeplane allows to specify a regular expression in the following places:
- `Edit->Find->Find and replace…` -> checkbox `Regular expressions`
- `Edit->Find->Find…` -> `Matches regexp`
- `View->Controls->Filter toolbar` -> `Matches regexp`
- `Filter->Compose filter` -> `Matches regexp`
- Map/Node Conditional Style definitions

**Reply by wordmuse** [reply]
Thank you very much indeed!

~ Bal

**Comment by kenfhill84083** [comment]
Have you tried the chatGPT Freeplane Tutor : https://chatgpt.com/g/g-CkkQa1aEP-freeplane-tutor ?

It has guided me thru some groovy scripts (about which I know next to nothing) and other stuff I used to ask about on the forum - and then wait and hope for answers.  It feels like a bit like a forum back-and-forth but more immediate and cut-to-the-chase.

(And it always compliments me on my prompts and feedback.  No matter how stupid my prompts are.)
